### ISSUE TITLE
Support enabling sanitizers easily

### ISSUE THREAD

**Issue by jdm** [servo/servo#27190]
Based on https://doc.rust-lang.org/nightly/unstable-book/compiler-flags/sanitizer.html, I have successfully compiled Servo by adding `-Zsanitizer=address` to the `rustflags` field in my .servobuild, and using the following diff:
```diff
diff --git a/python/servo/build_commands.py b/python/servo/build_commands.py
index 58ef42e68a..f2c9b0dc4a 100644
--- a/python/servo/build_commands.py
+++ b/python/servo/build_commands.py
@@ -645,6 +645,9 @@ class MachCommands(CommandBase):
         # given on the command line can override it
         opts = ["-Ztimings=info"] + opts

+        if "-Zsanitizer" in env.get("RUSTFLAGS", ""):
+            opts += ["-Zbuild-std", "--target", target or host]
+
         if very_verbose:
             print(["Calling", "cargo", "build"] + opts)
             for key in env:
```
If this turns out to be useful, it might be nice to add .servobuild support for enabling a sanitizer.

**Comment by jdm** [comment]
There's a subtlety that took me forever to discover - since the --target flag is required, the binary ends up in target/[target]/servo, so `./mach run` ends up running the wrong binary.

**Comment by ede1998** [comment]
@jdm Just stumbled over this issue. Maybe this can be resolved with #31429? Or would you also be interested to have other sanitizers easily available?

**Comment by jdm** [comment]
It would be interesting to see if UBsan works in servo, but we can track that in another issue.
